lib/elixir/lib/enum.ex

```diff
@@ -2259,14 +2266,12 @@ defmodule Enum do
     do: []
   def take_random(first..first, count) when is_integer(count) and count >= 1,
     do: [first]
-  def take_random(first..last, 1) when first > last,
-    do: take_random(last..first, 1)
-  def take_random(first..last, 1),
-    do: [random_index(last - first) + first]
+  def take_random(_.._ = range, 1),
+    do: [random(range)]
```

Comment:
Oh, good catch. So let's revert these changes for now and add a test. We
can discuss Enum.random/2 separately.

On Monday, July 4, 2016, Eksperimental <notifications@github.com> wrote:

> In lib/elixir/lib/enum.ex
> <https://github.com/elixir-lang/elixir/pull/4927#discussion_r69400684>:
>
> > @@ -2259,14 +2266,12 @@ defmodule Enum do
> >      do: []
> >    def take_random(first..first, count) when is_integer(count) and count >= 1,
> >      do: [first]
> > -  def take_random(first..last, 1) when first > last,
> > -    do: take_random(last..first, 1)
> > -  def take_random(first..last, 1),
> > -    do: [random_index(last - first) + first]
> > +  def take_random(_.._ = range, 1),
> > +    do: [random(range)]
>
> sorry, not unique, but if the given enumerable has unique values, it will
> never return repeated values
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/elixir-lang/elixir/pull/4927/files/8572da663e212f709689d504fa954f315bedd7bd#r69400684>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAAlbij6qxPBk9u4RzINKtNB69MVOVhTks5qSF5LgaJpZM4JD0MS>
> .
>


-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*